pkg/services/ngalert/notifier/silences.go

```diff
@@ -75,6 +76,14 @@ func (am *Alertmanager) GetSilence(silenceID string) (apimodels.GettableSilence,
 
 // CreateSilence persists the provided silence and returns the silence ID if successful.
 func (am *Alertmanager) CreateSilence(ps *apimodels.PostableSilence) (string, error) {
+	// Validate our silence to avoid segfaults
```

Comment:
I found this problem while writing tests, because I didn't know I had to set all the fields. I was causing segfaults, even. See: https://github.com/prometheus/alertmanager/issues/2851

If we validate somewhere further upstream, we can drop this.

Reply:
I think this validation needs to be done on the controller level for better separation of API\DTO layers. We can address it later.

Reply:
Great catch! I would split this into a separate PR so that we can discuss this individually.